src/ucc/api/ucc.h

```diff
@@ -1696,7 +1696,13 @@ typedef enum {
     UCC_COLL_ARGS_FLAG_IN_PLACE             = UCC_BIT(0), /*!< If set, the
                                                             output buffer is
                                                             identical to the
-                                                            input buffer.*/
+                                                            input buffer.
+                                                            Note, for inplace
+                                                            reduce scatter
```

Comment:
I think we need to document that somehow differently. Firstly, why "offset according to team id" ? did you mean to say "according to rank", but we don't really have "rank" defined ... Secondly, need to clearly defined what is the "offset" = rank * dst.info.count/team_size * dt_size. Thirdly, same applies to reduce-scatterv but offset is dst.info_v.displacements[rank] * dt_size. Maybe we should add all this to manual but not put here in ucc.h (looks too much) ?? @manjugv

Reply:
> I think we need to document that somehow differently. Firstly, why "offset according to team id" ? did you mean to say "according to rank", but we don't really have "rank" defined ... 

right, we don't have rank defined, that's why "according to team id"

Reply:
"team id" is not defined either. i suggest we move detailed description to manual.

Reply:
Or lets write:
default INPLACE behavior for UCC_COLL_TYPE_REDUCE_SCATTER and UCC_COLL_TYPE_REDUCE_SCATTERV is to place result into destination buffer at the offset rank * dst.info.count/team_size * dt_size or dst.info_v.displacements[rank] * dt_size respectively (where "rank" is the integer id of a process defined during team creation).

@manjugv plz help, i know you wanted to avoid using "rank" term but now it is very difficult to even write specification of the collective behavior without that term. For example we have info_v.counts/displacements arrays that are indexed using ranks of processes (not enpoint values), but we never defined rank itself.

Reply:
@Sergei-Lebedev @vspetrov ep=rank, when ep is ordered and contiguous. 

For reduce_scatter, we can write like this. If we agree on this, we can write to the rest of the collectives.

**The UCC_COLL_TYPE_REDUCE_SCATTER first performs a data element-wise reduction on the src buffer and then scatters the result to the dst buffer. If the size of src buffer is “count * dt_size”, where dt_size is the number of bytes for the data type element and count is the number of elements of that datatype. Assuming the result can be divided into “n” blocks, the ith block is placed in the receive buffer of endpoint “I”. Like other collectives, for this collective, the “ep” should be ordered and contiguous. 

When INPLACE is set for UCC_COLL_TYPE_REDUCE_SCATTER, the receive buffers act as both send and receive buffer.**

Reply:
Sounds good to me